lib/typescript/websocketclient/index.ts

```diff
@@ -0,0 +1,91 @@
+import {StompWrapper} from './stompWrapper';
+import {Message, Channel} from 'httpclient';
+import {messageMapper} from 'httpclient/mappers/messageMapper';
+import {channelMapper} from 'httpclient/mappers/channelMapper';
+
+type CallbackMap = {
+  onMessage?: (conversationId: string, channelId: string, message: Message) => void;
+  onUnreadCountUpdated?: (conversationId: string, unreadMessageCount: number) => void;
+  onChannelConnected?: (channel: Channel) => void;
+  onChannelDisconnected?: (channel: Channel) => void;
+  onError?: () => void;
+};
+
+export class WebSocketClient {
+  public readonly token?: string;
+  public readonly apiUrlConfig?: string;
+
+  stompWrapper: StompWrapper;
+  callbackMap: CallbackMap;
+
+  constructor(token: string, callbackMap: CallbackMap, baseUrl: string) {
+    this.token = token;
+    this.callbackMap = callbackMap;
+    this.apiUrlConfig = `ws://${baseUrl}/ws.communication`;
+
+    this.stompWrapper = new StompWrapper(
+      this.apiUrlConfig,
+      {
+        '/queue/message': item => {
```

Comment:
cannot we type the item? honest question

Reply:
They are typed in the StompWrapper class.

Reply:
ah nice